app/models/warehouse.rb

```diff
@@ -1,2 +1,4 @@
 class Warehouse < ApplicationRecord
+  validates :name, presence: true
+  validates :code, uniqueness: true, presence: true
```

Comment:
Acho que talvez o presence do `code` poderia ficar na mesma linha do name ou fazer alguma coisa assim pra deixar mais legível

```suggestion
  with_options presence: true do
    validates :name
    validates :code, uniqueness: true
  end
```